src/libs/ReportUtils.ts

```diff
@@ -641,163 +818,117 @@ function isAllowedToComment(report) {
     // If the user is an admin, allow them to post.
-    const policy = allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`];
-    return lodashGet(policy, 'role', '') === CONST.POLICY.ROLE.ADMIN;
+    const policy = allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`];
+    return policy?.role === CONST.POLICY.ROLE.ADMIN;
 }
 
 /**
  * Checks if the current user is the admin of the policy given the policy expense chat.
- * @param {Object} report
- * @param {String} report.policyID
- * @param {Object} policies must have OnyxKey prefix (i.e 'policy_') for keys
- * @returns {Boolean}
  */
-function isPolicyExpenseChatAdmin(report, policies) {
+function isPolicyExpenseChatAdmin(report: OnyxEntry<Report>, policies: OnyxCollection<Policy>): boolean {
     if (!isPolicyExpenseChat(report)) {
         return false;
     }
 
-    const policyRole = lodashGet(policies, [`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`, 'role']);
+    const policyRole = policies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`]?.role;
 
     return policyRole === CONST.POLICY.ROLE.ADMIN;
 }
 
 /**
  * Checks if the current user is the admin of the policy.
- * @param {String} policyID
- * @param {Object} policies must have OnyxKey prefix (i.e 'policy_') for keys
- * @returns {Boolean}
  */
-function isPolicyAdmin(policyID, policies) {
-    const policyRole = lodashGet(policies, [`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, 'role']);
+function isPolicyAdmin(policyID: string, policies: OnyxCollection<Policy>): boolean {
+    const policyRole = policies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`]?.role;
 
     return policyRole === CONST.POLICY.ROLE.ADMIN;
 }
 
 /**
  * Returns true if report has a single participant.
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function hasSingleParticipant(report) {
-    return report && report.participantAccountIDs && report.participantAccountIDs.length === 1;
+function hasSingleParticipant(report: OnyxEntry<Report>): boolean {
+    return report?.participantAccountIDs?.length === 1;
 }
 
 /**
  * Checks whether all the transactions linked to the IOU report are of the Distance Request type
  *
- * @param {string|null} iouReportID
- * @returns {boolean}
  */
-function hasOnlyDistanceRequestTransactions(iouReportID) {
+function hasOnlyDistanceRequestTransactions(iouReportID: string | undefined): boolean {
     const allTransactions = TransactionUtils.getAllReportTransactions(iouReportID);
-    return _.all(allTransactions, (transaction) => TransactionUtils.isDistanceRequest(transaction));
+    return allTransactions.every((transaction) => TransactionUtils.isDistanceRequest(transaction));
 }
 
 /**
  * If the report is a thread and has a chat type set, it is a workspace chat.
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isWorkspaceThread(report) {
+function isWorkspaceThread(report: OnyxEntry<Report>): boolean {
     return Boolean(isThread(report) && !isDM(report));
```

Comment:
```suggestion
    return isThread(report) && !isDM(report);
```
I don't think we need Boolean here